saleor/graphql/discount/mutations/sale/sale_channel_listing_update.py

```diff
@@ -73,18 +82,75 @@ class Meta:
         error_type_field = "discount_errors"
 
     @classmethod
-    def add_channels(cls, sale: "SaleModel", add_channels: List[Dict]):
+    def add_channels(cls, promotion: Promotion, add_channels: List[Dict]):
+        rules_data_to_add: List[RuleInfo] = []
+        rules_to_update: List[PromotionRule] = []
+
+        rules = promotion.rules.all()
+        PromotionRuleChannel = PromotionRule.channels.through
+        rule_ids = [rule.id for rule in rules]
+        rule_channel = PromotionRuleChannel.objects.filter(
+            promotionrule_id__in=rule_ids
+        )
+        current_channel_ids = rule_channel.values_list("channel_id", flat=True)
+        examplart_rule = rules[0]
```

Comment:
Why do we need that?

Reply:
In order to keep `catalogue_predicate` and `reward_value_type` and populate it to all rules.